pkg/core/native/native_neo.go

```diff
@@ -932,6 +932,9 @@ func (n *NEO) VoteInternal(ic *interop.Context, h util.Uint160, pub *keys.Public
 	}
 	ic.DAO.PutStorageItem(n.ID, key, acc.Bytes(ic.DAO.GetItemCtx()))
 
+	if pub == nil {
+		acc.LastGasPerVote = *big.NewInt(0)
+	}
```

Comment:
You should put this right before line 933 because `ic.DAO.PutStorageItem` is exactly the method that puts the updated account state into contract storage.